lib/dicom_net/association.ex

```diff
@@ -109,6 +111,60 @@ defmodule DicomNet.Association do
     {new_state, response}
   end
 
+  defp reject_association(state, association_data) do
+    new_state =
+      state
+      |> Map.put(:state, :association_release_request)
+      |> Map.put(:association, association_data)
+
+    response =
+      Pdu.new_association_release_response_pdu()
+      |> Pdu.serialize()
+
+    {new_state, response}
+  end
+
+  defp handle_pdu(
+         %Pdu{
+           type: :associate_request,
+           data: associate_request
+         },
+         %{socket: _socket, state: :waiting_for_association} = state
+       ) do
+    Logger.info("Accepting association")
+
+    # If the :association handler is defined 
+    # the acceptance/rejection can be handled by
+    # the host application.
+    {new_state, response} =
+      case Keyword.fetch(state.handlers, :association) do
```

Comment:
A bit nitpicky, but I'm not 100% sure with calling this handler just `association`. The code which handles the whole "association" is this module, the handler is just about it's establishment.

How about `:association_authenticate` or something?

Reply:
What about `:association_validator` ?

Reply:
Would be OK for me :+1:  I'm not sure if it's 100% accurate, since for me "validating" is more something you do to something which already exists to check if it complies with some expectation. While this handler is executed to check if a particular association should come into existence at all. But I also cannot really think of something which is much better - so let's go with this.